maven/transform2.py

```diff
@@ -35,44 +39,46 @@ def extract_and_transform_lines(actual):
     return transformed_lines
 
-# Function to transform lines with correct spacing
 def transform_lines_with_space(original):
     lines = original.strip().split("\n")
     transformed_lines = []
+    root_seen = False
 
     for line in lines:
-        line = line.replace("[info]", "").replace("[S]", "").strip()
-        # Skip lines that contain only pipes and spaces or empty lines
-        if re.match('^[ |]+$', line) or line == "":
+        original_line = line  # for calculating indent
+        clean_line = line.replace("[info]", "").replace("[S]", "").strip()
+        if re.match(r'^[\|\+\-\s]+$', clean_line) or clean_line == "" or should_be_skipped(clean_line):
             continue
-        # Skip "evicted" dependencies as they are superseded by a different version
-        # This regex is short and simple because "evicted" is sometimes truncated
-        # When revising, recommended to check behavior and performance with a regex tester
-        # It will also skip lines ending with ... as they don't show a valid dependency, 
-        # for example: +-io.opentelemetry.semconv:opentelemetry-semconv-incubating:1...
-        if should_be_skipped(line):
-            continue
-        parts = line.split(':')
-        if len(parts) > 2:
-            # Check if parts[2] matches the pattern: whitespace + bracket + (e|ev|.)
-            # Example, the line: com.fasterxml.jackson.core:jackson-databind:2.12.7.1 (..
-            # will be discarded
-            # before it was transformed to: com.fasterxml.jackson.core:jackson-databind:jar:2.12.7.1 (..:compile
-            # See: CSE-1676 for more details
+
+        # Calculate indentation depth (each 2 spaces == 1 level)
+        indent_match = re.match(r'^([ |]+)\+-', line)
+        if indent_match:
+            indent_str = indent_match.group(1)
+            depth = indent_str.count('|') + indent_str.count('  ')
+        else:
+            depth = 0
+
+        # Clean the line from old tree characters
+        line_content = re.sub(r'^[\|\+\-\s]+', '', clean_line)
+
+        parts = line_content.split(':')
+        if len(parts) >= 3:
             if re.search(r'\s[\(](e|ev|\.)', parts[2]):
-                continue  # Discard this line
+                continue
             parts.insert(2, 'jar')
             parts.append('compile')
-            line = ':'.join(parts)
-        # These ensure that the spacing of the pipes and +- notations are correct/consistent
-        line = line.replace("+-", "+- ")
-        line = re.sub(r'\|\s+', "|  ", line)
-        # Add the line to the list if it's not empty
-        if line:
-            transformed_lines.append(line)
-    
+            formatted_line = ':'.join(parts)
+
+            if not root_seen:
+                transformed_lines.append(formatted_line)
+                root_seen = True
+            else:
+                prefix = "|  " * (depth - 1) + "+- " if depth > 0 else "+- "
+                transformed_lines.append(prefix + formatted_line)
```

Comment:
I _think_ the script assumes that:

- Lines with indents = child
- Lines without indents = root

So currently it's not picking up direct dependencies?

We could do something like the below to detect root-level:

```suggestion
           # Treat any line with +- not prefixed by a pipe as root-level
           if re.search(r'^\s*[\+\\]-', original_line) and not re.search(r'^\s*\|', original_line):
                prefix = "+- "

            else:
                  prefix = "|  " * (depth - 1) + "+- " if depth > 0 else "+- "

            transformed_lines.append(prefix + formatted_line)
```

@sebastianrevuelta what are your thoughts?